components/TodoList.js

```diff
@@ -0,0 +1,68 @@
+function TodoList(app, data) {
+  const $todoList = document.querySelector(".todo-list");
+
+  // 개선이 필요해 보인다. 삭제시 렌더가 작동 안함..
+  this.render = () => {
+    $todoList.innerHTML = data
+      .map((todo) => {
+        return `
+        <li class="${todo.completed ? `completed` : ``}" id=${todo.id}>
+          <div class="view">
+            <input class="toggle" type="checkbox" ${
+              todo.completed ? `checked` : ``
+            }/>
+            <label class="label">${todo.text}</label>
+            <button class="destroy"></button>
+          </div>
+          <input class="edit" value="${todo.text}" />
+        </li>`;
+      })
+      .join(""); // join 메서드 사용처 정리
+  };
+
+  // editMode, editTodo 따로 선언한 이유 : e가 다르다 (<label>, <input>)
+  this.editMode = (e) => {
+    if (e.target.className === "label") {
+      e.target.closest("li").classList.add("editing"); // closest 정리
+    }
+  };
+  $todoList.addEventListener("dblclick", this.editMode); // EL 위치 몰아서 통일 가능?
```

Comment:
이거는 TodoList 컴포넌트 내부에서만 쓰이는 이벤트 핸들링 함수니까 깔끔하게 작성 가능 할 수 있지 않을까요?

이렇게요. 

```suggestion
  $todoList.addEventListener("dblclick", (e) => {
    if (e.target.className === "label") {
      e.target.closest("li").classList.add("editing"); // closest 정리
    }
  });
```

아래있는 `editTodo` 랑, 딜리드, 체크 함수 모두 마찬가지라고 생각합니다! 

그냥 깔끔하게 바로 읽힐 수 있게 생각해보았어요 ㅎㅎ

Reply:
제가 함수랑 이벤트리스너를 분리하자는 강박이 살짝 있었네여..ㅎㅎ 훨 보기 좋아요